src/phoebus_boundaries/phoebus_boundaries.cpp

```diff
@@ -460,82 +470,90 @@ void ProcessBoundaryConditions(parthenon::ParthenonManager &pman) {
+      //                        "Only \"user\" and \"periodic\" allowed for
+      //                        parthenon/mesh/" +
+      //                            name);
 
-      const std::string bc = pman.pinput->GetOrAddString("phoebus", name, "outflow");
+      const std::string bc = pman.pinput->GetOrAddString("phoebus", name, "gr_outflow");
       if (bc == "reflect") {
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = reflect[d - 1][outer];
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "reflect",
+                                                  reflect[d - 1][outer]);
       } else if (bc == "polar") {
         PARTHENON_REQUIRE(d == 2, "Polar boundary conditions only supported in X2!");
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = polar[d - 1][outer];
-      } else if (bc == "outflow") {
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = outflow[d - 1][outer];
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "polar",
+                                                  polar[d - 1][outer]);
+      } else if (bc == "gr_outflow") {
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "gr_outflow",
+                                                  outflow[d - 1][outer]);
         if (d == 1) {
           if (outer == 0) {
             if (rad_method == "mocmc") {
-              pman.app_input
-                  ->swarm_boundary_conditions[parthenon::BoundaryFace::inner_x1] =
-                  Boundaries::SetSwarmNoWorkBC;
+              pman.app_input->RegisterSwarmBoundaryCondition(
+                  parthenon::BoundaryFace::inner_x1, "swarm_no_bc",
+                  &Boundaries::SwarmNoWorkBC);
             } else {
-              pman.app_input
-                  ->swarm_boundary_conditions[parthenon::BoundaryFace::inner_x1] =
-                  Boundaries::SetSwarmIX1Outflow;
+              pman.app_input->RegisterSwarmBoundaryCondition(
+                  parthenon::BoundaryFace::inner_x1, "swarm_outflow",
+                  &parthenon::BoundaryFunction::SwarmOutflowInnerX1);
             }
           } else if (outer == 1) {
             if (rad_method == "mocmc") {
-              pman.app_input
-                  ->swarm_boundary_conditions[parthenon::BoundaryFace::outer_x1] =
-                  Boundaries::SetSwarmNoWorkBC;
+              pman.app_input->RegisterSwarmBoundaryCondition(
+                  parthenon::BoundaryFace::outer_x1, "swarm_no_bc",
+                  &Boundaries::SwarmNoWorkBC);
             } else {
-              pman.app_input
-                  ->swarm_boundary_conditions[parthenon::BoundaryFace::outer_x1] =
-                  Boundaries::SetSwarmOX1Outflow;
+              pman.app_input->RegisterSwarmBoundaryCondition(
+                  parthenon::BoundaryFace::outer_x1, "swarm_outflow",
+                  &parthenon::BoundaryFunction::SwarmOutflowOuterX1);
```

Comment:
Same here. No need to condition anymore. We can remove the phoebus bc input option complete. Just register all our BCs and let parthenon handle the conditionals.